attributable_error_crypto.go

```diff
@@ -0,0 +1,103 @@
+package sphinx
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+)
+
+type payloadSource byte
+
+const (
+	// payloadIntermediateNode is a marker to signal that this attributable
+	// error payload is originating from a node between the payer and the error
+	// source.
+	payloadIntermediateNode payloadSource = 0
+
+	// payloadErrorNode is a marker to signal that this attributable error
+	// payload is originating from the error source.
+	payloadErrorNode payloadSource = 1
+)
+
+type AttributableErrorStructure struct {
+	HopCount        int
+	FixedPayloadLen int
+}
+
+type attributableErrorBase struct {
+	maxHops             int
```

Comment:
could make these returned by methods on the above `struct` instead. I think that is cleaner than pre-calculating constants.

Reply:
I gave that a try, but I don't think the addition of all the parenthesis for the new function calls make the code more readable.

Reply:
As you both don't seem to be too bothered with the parenthesis, I went ahead and did the conversion.